Add Theme Toggler plugin

larsmagnus opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/larsmagnus/obsidian-theme-toggler

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

liamcain commented:
- [buttonSettings: {](https://github.com/larsmagnus/obsidian-theme-toggler/blob/8fb14c29a90decac65f49bc84cf4eed126241ac3/src/constants.ts#L15-L19) this doesn't belong in the settings. Settings should only include data that's persisted across restarts.
- [await this.saveSettings(DEFAULT_SETTINGS)](https://github.com/larsmagnus/obsidian-theme-toggler/blob/8fb14c29a90decac65f49bc84cf4eed126241ac3/src/main.ts#L195)  I don't think you want to include this in onunload. If you don't want to persist the leaf configurations, I suggest you store them in an instance variable rather than settings.
- [getAppTheme](https://github.com/larsmagnus/obsidian-theme-toggler/blob/8fb14c29a90decac65f49bc84cf4eed126241ac3/src/main.ts#L167) This function call isn't doing anything
- [this.app.commands.executeCommandById(`${pluginId}:${id}`)](https://github.com/larsmagnus/obsidian-theme-toggler/blob/8fb14c29a90decac65f49bc84cf4eed126241ac3/src/main.ts#L41-L42) This doesn't look right. `id` here is coming from buttonSettings.id, right? This approach seems more complicated than it needs to be. Why not just have the function call `toggleTheme`? Having it call `executeCommandById` seems unnecessary.
- [mousedown](https://github.com/larsmagnus/obsidian-theme-toggler/blob/8fb14c29a90decac65f49bc84cf4eed126241ac3/src/main.ts#L38) I think this should be `'click'` instead of mousedown. You also don't need to use `registerDomEvent` here, you can just use `addEventListener('click')` here. When the plugin unloads, you are removing the button anyway.

larsmagnus commented:
Thanks for the thorough review @liamcain! 🙌

I've addressed all your points in the 1.0.1 release of the plugin. Let me know if there are any other steps I need to make as part of the review process, or if there's anything else that needs fixing.
